pkg/kube/secrets.go

```diff
@@ -165,36 +165,40 @@ func (r *secretsReader) ListByLocalObjectReferences(ctx context.Context, refs []
 	return secrets, nil
 }
 
-func (r *secretsReader) ListImagePullSecretsByPodSpec(ctx context.Context, spec corev1.PodSpec, ns string) ([]corev1.Secret, error) {
-	// try to fetch image pull secret from pod spec
-	secrets, err := r.ListByLocalObjectReferences(ctx, spec.ImagePullSecrets, ns)
-	if err != nil {
-		return nil, err
-	}
-
+func (r *secretsReader) getServiceAccountByPodSpec(ctx context.Context, spec corev1.PodSpec, ns string) (*corev1.ServiceAccount, error) {
 	serviceAccountName := spec.ServiceAccountName
 	if serviceAccountName == "" {
 		serviceAccountName = serviceAccountDefault
 	}
-	// try to fetch image pull secret from service account
-	var sa corev1.ServiceAccount
-	err = r.client.Get(ctx, client.ObjectKey{Name: serviceAccountName, Namespace: ns}, &sa)
+
+	sa := &corev1.ServiceAccount{}
+	err := r.client.Get(ctx, client.ObjectKey{Name: serviceAccountName, Namespace: ns}, sa)
 	if err != nil {
 		return nil, fmt.Errorf("getting service account by name: %s/%s: %w", ns, serviceAccountName, err)
 	}
-	serviceAccountSecrets, err := r.ListByLocalObjectReferences(ctx, sa.ImagePullSecrets, ns)
+	return sa, nil
+}
+
+func (r *secretsReader) ListImagePullSecretsByPodSpec(ctx context.Context, spec corev1.PodSpec, ns string) ([]corev1.Secret, error) {
+	imagePullSecrets := spec.ImagePullSecrets
+
+	sa, err := r.getServiceAccountByPodSpec(ctx, spec, ns)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Then this will not be a refactor, but I agree that the suggested change should be considered:

https://github.com/aquasecurity/trivy-operator/blob/4b708839670341280b3a65856573c6d65449e0c0/pkg/kube/secrets.go#L179-L183

Reply:
@chen-keinan Do you want me to change this behavior in this PR? I would prefer not to, since I planned for a clean refactoring. Maybe open an issue from this discussion?